databuilder/databuilder/extractor/neo4j_extractor.py

```diff
@@ -88,16 +103,21 @@ def _get_extract_iter(self) -> Iterator[Any]:
         """
         Execute {cypher_query} and yield result one at a time
         """
-        with self.driver.session() as session:
-            if not hasattr(self, 'results'):
-                self.results = session.read_transaction(self._execute_query)
-
-            for result in self.results:
-                if hasattr(self, 'model_class'):
-                    obj = self.model_class(**result)
-                    yield obj
-                else:
-                    yield result
+        db_name = self.conf.get(Neo4jExtractor.NEO4J_DATABASE_NAME, None)
+        if db_name:
```

Comment:
if i'm not mistaken, `neo4j.DEFAULT_DATABASE` can be used as the db_name if one is not passed in, eliminating this check.